src/api/app/controllers/webui/request_controller.rb

```diff
@@ -225,7 +227,26 @@ def changerequest
           flash[:error] = 'Will not add maintainer for not accepted requests'
         end
       end
-      accept_request if changestate == 'accepted'
+
+      # This is how we try to make the submitter a maintainer while using the beta request show page
+      if changestate == 'accepted'
+        if params[:accepted] == 'Accept and make the creator a maintainer of the target'
+          @bs_request.bs_request_actions.each do |action|
+            target = action.target_package_object || action.target_project_object
```

Comment:
What about new packages? We don't tell people that we will make the creator a maintainer of the project anywhere...

Reply:
What new packages?

Reply:
I can submit a *new* package to the target project. In that case `action.target_package_object` is nil.

Reply:
Okay, in that case we're going to make the creator of the request a maintainer of the project. But only if the approving user has permissions to do so... Maybe that's something we don't want?

Reply:
> Maybe that's something we don't want?

I don't think it is. We should make the create maintainer *after* we've accepted the request (which will create the package).